crates/biome_css_analyze/src/lint/nursery/use_sorted_properties.rs

```diff
@@ -0,0 +1,387 @@
+enum NodeKindOrder {
+    CustomProperty,
+    ComposesProperty,
+    Declaration,
+    UnknownDeclaration,
+    NestedRuleOrAtRule,
+    UnknownKind,
+}
+
+#[derive(PartialEq, Eq, PartialOrd, Ord, Debug)]
+/// Define sort order using lexographical sorting of this struct
+pub struct SortInfo {
+    // First, nodes are sorted by the kind of node (e.g. declarations before at-rules)
+    kind: NodeKindOrder,
+    // Property nodes are sorted by a predefined desired order
+    property: u32,
+    // Vendor prefixed properties should be before the non-prefixed version of the same property
+    vendor_prefix: u32,
+    original_position: usize,
+}
+
+impl SortInfo {
+    fn from_kind(kind: NodeKindOrder) -> Self {
+        Self {
+            kind,
+            property: 0,
+            vendor_prefix: 0,
+            original_position: 0,
+        }
+    }
+    fn from_declaration(property: u32, vendor_prefix: u32) -> Self {
+        Self {
+            kind: NodeKindOrder::Declaration,
+            property,
+            vendor_prefix,
+            original_position: 0,
+        }
+    }
+    fn unknown() -> Self {
+        Self {
+            kind: NodeKindOrder::UnknownKind,
+            property: 0,
+            vendor_prefix: 0,
+            original_position: 0,
+        }
+    }
+}
+
+impl From<&AnyCssDeclarationOrRule> for SortInfo {
+    fn from(node: &AnyCssDeclarationOrRule) -> SortInfo {
+        match node {
+            AnyCssDeclarationOrRule::CssEmptyDeclaration(_) => SortInfo::unknown(),
+            AnyCssDeclarationOrRule::CssBogus(_) => SortInfo::unknown(),
+            AnyCssDeclarationOrRule::CssMetavariable(_) => SortInfo::unknown(),
+            AnyCssDeclarationOrRule::AnyCssRule(rule) => match rule {
+                AnyCssRule::CssAtRule(_) => SortInfo::from_kind(NodeKindOrder::NestedRuleOrAtRule),
+                AnyCssRule::CssBogusRule(_) => SortInfo::unknown(),
+                AnyCssRule::CssNestedQualifiedRule(_) => {
+                    SortInfo::from_kind(NodeKindOrder::NestedRuleOrAtRule)
+                }
+                AnyCssRule::CssQualifiedRule(_) => SortInfo::unknown(),
+            },
+            AnyCssDeclarationOrRule::CssDeclarationWithSemicolon(decl_with_semicolon) => {
+                let prop = decl_with_semicolon
+                    .declaration()
+                    .ok()
+                    .and_then(|decl| decl.property().ok());
+
+                match prop {
+                    Some(AnyCssProperty::CssComposesProperty(_)) => {
+                        SortInfo::from_kind(NodeKindOrder::ComposesProperty)
+                    }
+                    Some(AnyCssProperty::CssGenericProperty(prop)) => match prop.name().ok() {
+                        Some(AnyCssDeclarationName::CssDashedIdentifier(_)) => {
+                            SortInfo::from_kind(NodeKindOrder::CustomProperty)
+                        }
+                        Some(AnyCssDeclarationName::CssIdentifier(ident)) => {
```

Comment:
nit: To reduce branching and avoid writing `Some()` every time, you can use `let/else` e.g. 

```rs
let Some(prop) = prop else {
	return SortInfo::unknown()
};

match prop {
	AnyCssProperty::CssComposesProperty(_) => {},
	_ => {} // add the rest
}
```